pkg/controller/persistentvolume/index.go

```diff
@@ -87,15 +89,29 @@ func (pvIndex *persistentVolumeOrderedIndex) findByClaim(claim *api.PersistentVo
 		unboundVolumes := []*api.PersistentVolume{}
 		for _, volume := range volumes {
 			// volume isn't currently bound or pre-bound.
-			if volume.Spec.ClaimRef == nil {
-				unboundVolumes = append(unboundVolumes, volume)
-				continue
-			}
-
 			if isVolumeBoundToClaim(volume, claim) {
```

Comment:
I can't add a unit test in index_test for this because it assumes that the volume name isn't set.  I'll look for the right place in the controller to get coverage for this.